db/migrate/20160301141911_duplicate_people_by_community_membership.rb

```diff
@@ -0,0 +1,113 @@
+class DuplicatePeopleByCommunityMembership < ActiveRecord::Migration
+  def up
+    people = select_values("
+               SELECT p.id
+               FROM people AS p
+               LEFT OUTER JOIN community_memberships AS cm
+               ON p.id = cm.person_id
+               GROUP BY p.id
+               HAVING count(cm.person_id) > 1
+             ")
+    people.each do |person_id|
+      memberships = select_all("
+        SELECT * FROM community_memberships WHERE person_id = #{quote(person_id)}
+      ").to_ary
+      unless memberships.empty?
```

Comment:
There's a change to optimize and simplify this:

If you change

```
SELECT p.id
FROM people AS p
LEFT OUTER JOIN community_memberships AS cm
ON p.id = cm.person_id
GROUP BY p.id
HAVING count(cm.person_id) > 1
```

to

```
SELECT p.id, count(cm.person_id)
FROM people AS p
LEFT OUTER JOIN community_memberships AS cm
ON p.id = cm.person_id
GROUP BY p.id
HAVING count(cm.person_id) > 1
```

...then we already know how many new people we need to insert and can skip the `SELECT * FROM community_memberships WHERE person_id = #{quote(person_id)}` query.